### PR TITLE
Added initial tests for createChallengeTasks and fixed some issues in…

### PR DESCRIPTION
**Pull request by TheHollidayInn**
I ran into some issues when trying to create challenge tasks for my group leave tests, so I fixed the errors and wrote some initial tests for the routes.

### CODE REVIEW

File: test/api/v3/integration/tasks/challenges/POST-tasks_challenge_id.test.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      notes: 1976,
+    })).to.eventually.be.rejected.and.eql({
+      code: 404,
+      error: 'NotFound',
+      message: t('challengeNotFound'),
+    });
+  });
+
+  xit('returns error when non leader tries to edit challenge', async() => {
+    // @TODO needs join chellenge route
```
Comment (paglias) [inline test/api/v3/integration/tasks/challenges/POST-tasks_challenge_id.test.js]: @TheHollidayInn we could just populate the non-leader `challenges` array, or not?

**Comment by paglias** [comment]
@crookedneighbor here do we want to mirror user's tasks tests as far as possible?

**Comment by crookedneighbor** [comment]
I'm inclined to say it should mirror, but only because the test coverage is pretty thorough. If the coverage is thorough here, that's fine.

File: test/api/v3/integration/tasks/challenges/POST-tasks_challenge_id.test.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  beforeEach(async () => {
+    user = await generateUser({balance: 1});
+    guild = await user.post('/groups', {
+      name: 'Test Public Guild',
+      type: 'guild',
+    });
+    challenge = await user.post('/challenges', {
+      name: 'Test Challenge',
+      shortName: 'TChallenge',
+      leader: user._id,
```
Comment (crookedneighbor) [inline test/api/v3/integration/tasks/challenges/POST-tasks_challenge_id.test.js:21]: This param gets ignored in the route. The leader is assigned by the user making the request.

**Comment by crookedneighbor** [comment]
Merged in with d93cabafb950b0e2d2c1e548cfa45bcb322118ef

**Comment by crookedneighbor** [comment]
s/closed/merged in

**Pull request closed without merging**